lib/semantics/resolve-names.cc

```diff
@@ -1757,7 +1760,10 @@ static bool NeedsType(const Symbol &symbol) {
 }
 void ScopeHandler::ApplyImplicitRules(Symbol &symbol) {
   if (NeedsType(symbol)) {
-    if (isImplicitNoneType()) {
+    if (const auto *type{GetImplicitType(symbol)}) {
+      symbol.set(Symbol::Flag::Implicit);
+      symbol.SetType(*type);
+    } else {
```

Comment:
You could make this an `if-else-if` chain instead.

Reply:
Thanks, done.